Updated Scripts from Jackson

xmpro-dk opened the pull request:
Created on behalf of Jackson.

xmpro-dk commented:
@jaun-van-heerden can you review this? Can you also work with @jacksonmcguirkxmpro and @Kimmarie52 to first review the external content that it creates before it is synced up to the XMPro Platform space.

xmpro-dk commented:
If we don't already have, can we get somewhere documented on how the process works for ensuring new external content is created and synced up to GitBook (for Kim).

jaun-xmpro commented:
> @jaun-van-heerden can you review this? Can you also work with @jacksonmcguirkxmpro and @Kimmarie52 to first review the external content that it creates before it is synced up to the XMPro Platform space.

Sure thing :)

jaun-xmpro commented on scripts/xmpro-learning-yt-transcript-scripts/scrape-xmpro-learning-yt-transcrcipt-config.json:
```diff
@@ -0,0 +1,4 @@
+{
+    "folderPath": "docs/external-content/xmpro-learning-youtube",
```
@jacksonmcguirkxmpro 

[same as here](https://github.com/XMPro/public-docs/pull/6/commits/3c24a90efc3a34d8def371afdd133d9f8e1aad97#diff-5197550e5489be215d608fc6401370e7687e76d3a7886f836fafe835a77c443fR2)

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-about-config.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "folderPath": "docs/external-content/about"
```
@jacksonmcguirkxmpro 

[same as here](https://github.com/XMPro/public-docs/pull/6/commits/3c24a90efc3a34d8def371afdd133d9f8e1aad97#diff-5197550e5489be215d608fc6401370e7687e76d3a7886f836fafe835a77c443fR2)

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-blogs-config.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "folderPath": "docs/external-content/blogs"
```
@jacksonmcguirkxmpro 

[same as here](https://github.com/XMPro/public-docs/pull/6/commits/3c24a90efc3a34d8def371afdd133d9f8e1aad97#diff-5197550e5489be215d608fc6401370e7687e76d3a7886f836fafe835a77c443fR2)

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-latestnews-config.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "folderPath": "docs/external-content/latest-news"
```
@jacksonmcguirkxmpro 

[same as here](https://github.com/XMPro/public-docs/pull/6/commits/3c24a90efc3a34d8def371afdd133d9f8e1aad97#diff-5197550e5489be215d608fc6401370e7687e76d3a7886f836fafe835a77c443fR2)

TODO for all

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-latestnews.py:
```diff
@@ -0,0 +1,165 @@
+            readme_content.append(f"* [{file_info['title']}]({file_path})\n")
+        
+        # Create the README.md file in the same folder as the exported files
+        readme_file_path = os.path.join(folder_path, 'copy-me-latest-news.md')
```
@jacksonmcguirkxmpro 

lets call all of these `copy-me.md` so that they are easy to search for all of them at once

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-latestnews.py:
```diff
@@ -0,0 +1,165 @@
+                if filename:
+                    exported_files.append({'title': title, 'path': filename})
+            else:
+                print(f"Failed to scrape the page: {url}")
```
@jacksonmcguirkxmpro 

Lets consider what happens if `scrape_page(url) fails` we might want to either allow the `scape_page()` function to gracefully fail (not suddenly throw error, but instead return `False` or something) or wrap it up in a `try-except` statement to catch errors and log - then continue

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-usecases.py:
```diff
@@ -0,0 +1,147 @@
+import os
+import requests
+from bs4 import BeautifulSoup
+from pathvalidate import sanitize_filename
```
@jacksonmcguirkxmpro 

I see we are using a library here but previously did some find replaces - lets figure the best option and then make consistent

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-usecases.py:
```diff
@@ -0,0 +1,147 @@
+                            prop = "src"
+                            if "data-src" in element.attrs:
+                                prop = "data-src"
+                            w = element.get("width")
```
@jacksonmcguirkxmpro 

Lets consider the case if width or any other attribute doesn't exist

jacksonmcguirkxmpro replied:
If the width or other attributes do not exit, rather then stopping the code or not showing the image, it continues to show the images but at its original width.

jaun-xmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-usecases.py:
```diff
@@ -0,0 +1,147 @@
+                # Get the filename without the folder path
+                filename = os.path.basename(md_file['filename'])
+                # Get the folder path location without the "docs/" part and replace "\" with "/"
+                formatted_filename = md_file['filename'].replace("docs/", "").replace("\\", "/")
```
@jacksonmcguirkxmpro 

Instead of find replace - lets turn this into a `Path` object and then get the correct path - 

We can run an exercise on this - just let me know when you are free

jacksonmcguirkxmpro replied:
resoleved all scripts with path.as_posix(_) and path.relative("docs/")

jaun-xmpro reviewed (changes requested):
Great stuff @jacksonmcguirkxmpro - 
Some solid code here - have a read through here, make any obvious and quick changes 
Some comments are on certain lines but please scan the scripts to ensure the same issue/amendment is addressed everywhere.

Then we can catch up, talk through - identify the critical point, log the must do's and the nice-to-haves - run through some short exercises to churn some of the concepts

jaun-xmpro commented:
@Kimmarie52 

cc. @jacksonmcguirkxmpro 

Could you please suggest the required format in markdown or we could discuss and then just set some rules to follow for consistent formatting?

jacksonmcguirkxmpro commented on scripts/test-script.py:
```diff
@@ -0,0 +1,32 @@
+
+# Provide your username and password here
+username = "jackson.mcgurik@xmpro.com"
+password = "$Jstark2004"
```
Yes, Sorry this was a testing script, and I have removed it now.

jacksonmcguirkxmpro commented on scripts/scrape-xmpro-yt-transcripts-config.json:
```diff
@@ -1,4 +1,4 @@
 {
-    "folderPath": "docs/resources/external-content/youtube",
+    "folderPath": "docs/external-content/youtube",
```
Yes I agree, I did not realize that the path had changed, will update the config files.

xmpro-dk commented:
Can we please hold off merging this until I have fixed the issue which is causing some pages modified in GitBook, to have all their links updated and files move to the root GitHub Repo?
I have added a .gitbook.yaml file with config to fix the issue and am fixing recent updates from GitBook in GitHub that went awry.
https://github.com/XMPro/public-docs/pull/17

jacksonmcguirkxmpro commented on scripts/xmpro-website-scripts/scrape-xmpro-website-about.py:
```diff
@@ -0,0 +1,86 @@
+def save_to_md(title, content, url, folder_path):
+    try:
+        # Format the title to use as filename
+        filename_title = title.strip().lower().replace(" ", "-")
```
Work in progress some page titles have spaces - space, which then trick it to outputting --- .

jacksonmcguirkxmpro commented on scripts/xmpro-learning-yt-transcript-scripts/xmpro-learning-yt-transcript.py:
Completed

jacksonmcguirkxmpro replied:
Runs through the list of provided channels within json file and scrapes transcripts
